packages/apputils/src/toolbar.tsx

```diff
@@ -554,48 +550,37 @@ namespace Private {
    * This wraps the ToolbarButtonComponent and watches the kernel
    * session for changes.
    */
-  export class KernelNameComponent extends React.Component<
-    KernelNameComponent.IProps
-  > {
-    constructor(props: KernelNameComponent.IProps) {
-      super(props);
-      props.session.kernelChanged.connect(
-        this._onKernelChanged,
-        this
-      );
-      this._childProps = {
-        className: TOOLBAR_KERNEL_NAME_CLASS,
-        onClick: () => {
-          this.props.session.selectKernel();
-        },
-        tooltip: 'Switch kernel',
-        label: props.session.kernelDisplayName
-      };
-    }
 
-    public render() {
-      return <ToolbarButtonComponent {...this._childProps} />;
-    }
-
-    /**
-     * Update the text of the kernel name item.
-     */
-    private _onKernelChanged(session: IClientSession): void {
-      this._childProps.label = session.kernelDisplayName;
-      this.forceUpdate();
-    }
-
-    private _childProps: ToolbarButtonComponent.IProps;
+  export function KernelNameComponent(props: KernelNameComponent.IProps) {
+    return (
+      <UseSignal signal={props.session.kernelChanged} sender={props.session}>
+        {session => (
+          <ToolbarButtonComponent
+            className={TOOLBAR_KERNEL_NAME_CLASS}
+            onClick={props.session.selectKernel.bind(props.session)}
+            tooltip={'Switch kernel'}
+            label={session.kernelDisplayName}
+          />
+        )}
+      </UseSignal>
+    );
   }
 
   /**
-   * Phosphor Widget version of ToolbarButtonComponent.
+   * Phosphor widget version of KernelNameComponent.
    */
-  export class KernelName extends ReactElementWidget {
+  export class KernelName extends ReactWidget {
     constructor(props: KernelNameComponent.IProps) {
```

Comment:
Make this a react element widget, then can remove `render` method

Reply:
Agreed